src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -12,24 +12,24 @@ class OrderedRVDPartitioner(
   // rangeBounds: Array[Interval[kType]]
   // rangeBounds is interval containing all keys within a partition
   val rangeBounds: IndexedSeq[Interval],
-  val allowedOverlap: Int
+  allowedOverlap: Int
 ) {
   def this(
-    partitionKey: Array[String],
     kType: TStruct,
     rangeBounds: IndexedSeq[Interval]
-  ) = this(kType, rangeBounds, math.max(partitionKey.length - 1, 0))
+  ) = this(kType, rangeBounds, kType.size)
 
   def this(
-    partitionKey: Option[Int],
+    partitionKey: Array[String],
     kType: TStruct,
     rangeBounds: IndexedSeq[Interval]
-  ) = this(kType, rangeBounds, partitionKey.map(_ - 1).getOrElse(kType.size))
+  ) = this(kType, rangeBounds, math.max(partitionKey.length - 1, 0))
```

Comment:
Not clear to me why this is length - 1. Is it because `allowedOverlap = partitionKey - 1`?

Reply:
Right, `allowedOverlap = partitionKey.length - 1` is the translation between the old semantics and the new. As an example, consider if there are 3 key fields, and the partition key consists of the first two key fields. This means that we don't allow a single partition key to occur in more than one partition. In the new language, this is equivalent to saying that keys from two different partitions can agree in at most the first key field.

The reason for the change from `partitionKey` to `allowedOverlap` is to allow the case `allowedOverlap=key.length`, which has no analogue in the old language. This is the case where we allow a key to occur in multiple partitions (as long as we can guarantee that keys are globally ordered across partitions). The most basic case of this is when `allowedOverlap = key.length = 0`, in which case there is only one key value---`Row()`, and it is allowed to occur in multiple partitions, so this case subsumes `UnpartitionedRVD`.